axum/src/extract/extension.rs

```diff
@@ -73,3 +75,12 @@ impl<T> Deref for Extension<T> {
         &self.0
     }
 }
+
+impl<T> IntoResponseParts for Extension<T>
+where
+    T: Send + Sync + 'static,
+{
+    fn into_response_parts(self, res: &mut ResponseParts) {
+        res.insert_extension(self.0);
+    }
+}
```

Comment:
I'm thinking if we should even implement `tower::Layer` for `Extension`? Then we wouldn't need `AddExtension` layer and users would only deal with `Extension`. But maybe thats odd? 🤔

Reply:
Oohh, nice idea! I don't really see why not. I'm personally going to continue using `ServiceBuilderExt::add_extension` either way, but as this would reduce axum's API surface a little bit, I'd be in favor.